recipes/paho-mqtt-cpp/all/conanfile.py

```diff
@@ -44,13 +44,21 @@ def configure(self):
         self.options["paho-mqtt-c"].shared = self.options.shared
         self.options["paho-mqtt-c"].ssl = self.options.ssl
 
+
     def requirements(self):
-        self.requires("paho-mqtt-c/1.3.0")
+        if self.version == "1.1":
+            self.requires("paho-mqtt-c/1.3.1")
+        elif self.version == "1.0.1":
+            self.requires("paho-mqtt-c/1.3.0")
+        else:
+            raise ConanInvalidConfiguration("{} requirements not implemented".format(self.version))
 
     def source(self):
         tools.get(**self.conan_data["sources"][self.version])
         extracted_dir = self.name.replace("-", ".") + "-" + self.version
         os.rename(extracted_dir, self._source_subfolder)
+        for patch in self.conan_data["patches"][self.version]:
```

Comment:
do not apply patches in source, keep it there in build. Why? Source should be clean, any there will result in a new download. Also, if you need to check the author's code, without modifications, won't be possible.

Reply:
ok, will do

Reply:
thanks btw for the explaination, I already wondered why they are in build, 
however, just an other incompatibility with the developer flow from the manual, because there I want the source patched .....